src/msw/overlay.cpp

```diff
@@ -0,0 +1,227 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/msw/overlay.cpp
+// Author:      Ali Kettab
+// Copyright:   (c) 2022 Ali Kettab
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#include "wx/wxprec.h"
+
+#include "wx/private/overlay.h"
+#include "wx/dcclient.h"
+#include "wx/nativewin.h"
+#include "wx/window.h"
+#include "wx/sysopt.h"
+#include "wx/msw/private.h"
+
+namespace // anonymous
+{
+static const wxChar* gs_overlayClassName = nullptr;
+
+class wxOverlayWindow : public wxNativeContainerWindow
+{
+public:
+    static wxWindow*  Create(bool fullScreen = false)
```

Comment:
Do we use, or plan to use, `fullScreen` parameter? If not, I'd rather remove it for clarity.

Also, and less importantly, wx typically uses `Create()` for non-static ctor-like function, so I'd rename this to `New()`, which is used for similar factory functions in a couple of other classes.